src/epcpm/sunspectointerface.py

```diff
@@ -0,0 +1,2227 @@
+                "{\n",
+            ]
+        )
+        register_index = 0
+        rw_index = 0
+        # Account for 'SunS' length.
+        addr_index = epcpm.pm_helper.SUNS_LENGTH
+        for block_points in model_points:
+            for point in block_points:
+                point_name = (
+                    f"SUNSPEC{self.sunspec_id.value}_MODEL{point.model_id}_{point.name}"
+                )
+                first_reg = f"&sunspec{self.sunspec_id.value}Regs[{register_index}]"
+
+                if point.type == "pad16":
+                    getter = "NULL"
+                else:
+                    getter = f"&get{point_name}"
+
+                if point.read_write == "RW":
+                    total_rw_registers += point.size
+                if point.type == "SunspecModelHeader":
+                    addr_index += point.size
+                    continue
+                if point.size > 1:
+                    for point_size_index in range(point.size):
+                        point_full_name = f"{point_name}{point_size_index}"
+                        if point.read_write == "RW":
+                            setter = f", .w = &sunspec{self.sunspec_id.value}RegsRw[{rw_index}]"
+                            rw_index += 1
+                        else:
+                            setter = ""
+                        if point_size_index > 0:
+                            getter = "NULL"
+                        c_lines.extend(
+                            [
+                                f"    [{register_index}] = MODBUS_REGISTER_DEFAULTS(/* {point_full_name: <{longest_point_name_len}} */ .addr = {addr_index}, .firstReg = {first_reg}, .size = {point.size}, .get = {getter}{setter}),\n"
+                            ]
+                        )
+                        register_index += 1
+                        addr_index += 1
+
+                else:
+                    if point.read_write == "RW":
+                        setter = (
+                            f", .w = &sunspec{self.sunspec_id.value}RegsRw[{rw_index}]"
+                        )
+                        rw_index += 1
+                    else:
+                        setter = ""
+                    c_lines.extend(
+                        [
+                            f"    [{register_index}] = MODBUS_REGISTER_DEFAULTS(/* {point_name: <{longest_point_name_len}} */ .addr = {addr_index}, .firstReg = {first_reg}, .size = {point.size}, .get = {getter}{setter}),\n"
+                        ]
+                    )
+                    register_index += 1
+                    addr_index += 1
+        c_lines.extend(
+            [
+                "};\n\n",
+            ]
+        )
+
+        c_lines.extend(
+            [
+                f"ModbusReg * const sunspec{self.sunspec_id.value}AddrRegMap[{total_addresses}] =\n",
+                "{\n",
+            ]
+        )
+        # Account for 'SunS' length.
```

Comment:
This block has the same comment as the one starting on line 350, but this one is doing far less. Seems like both can be extracted out into a function that you use a param to indicate when you need the extra steps?

Reply:
The comment is only for the following line. I could go further, but abstracting out also makes it a little harder to maintain.